File: backend/models/incentiveTypeModel.js
import mongoose from 'mongoose';

const incentiveTypeSchema = new mongoose.Schema(
  {
    company: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Company',
      required: true,
      index: true,
    },
    name: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      trim: true,
      default: "",
    },
    isTaxable: {
      type: Boolean,
      default: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

// Ensure names are unique per company
incentiveTypeSchema.index({ company: 1, name: 1 }, { unique: true });

const IncentiveType = mongoose.model('IncentiveType', incentiveTypeSchema);
export default IncentiveType;

File: frontend/lib/models/incentive_type_model.dart
class IncentiveType {
  final String id;
  final String name;
  final String description;
  final bool isTaxable;
  final bool isActive;

  IncentiveType({
    required this.id,
    required this.name,
    required this.description,
    required this.isTaxable,
    required this.isActive,
  });

  factory IncentiveType.fromJson(Map<String, dynamic> json) {
    return IncentiveType(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      isTaxable: json['isTaxable'] ?? true,
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'isTaxable': isTaxable,
      'isActive': isActive,
    };
  }
}
